react/ChallengeTradePolicyCondition.tsx

```diff
@@ -95,30 +111,45 @@ const ChallengeTradePolicyCondition: FC<Props> = ({
   const isUnauthorized = isSessionUnauthorized(sessionResponse)
   const isForbidden = isSessionForbidden(sessionResponse)
   const profileCondition = isProfileAllowed(sessionResponse)
+  const tradePolicyCondition = isTradePolicyAllowed(sessionResponse)
+  const actionArgs = {returnUrl: redirectUnauthorizedTradePolicy}
+  const [logout] = useRedirectLogout({actionArgs})
 
   useRedirect(
     isUnauthorized === true || profileCondition === 'unauthorized',
     redirectPath
   )
+
+
   useRedirect(
     isForbidden === true || profileCondition === 'forbidden',
     forbiddenRedirectPath
   )
 
+
+  if(tradePolicyCondition){
+    logout()
+  }
```

Comment:
Pra chamar side effects dentro de um componente, usa o useEffect
```suggestion
useEffect(() => {
 if(tradePolicyCondition){
    logout()
  }
}, [logout, tradePolicyCondition] )

```